### ISSUE TITLE
ordered map problems with update-in

### ISSUE THREAD

**Issue by LispEngineer** [amalloy/ordered#31]
Hi there,

It appears that there is a problem with ordered 1.5.5.

`test1` is the ordered test data, and `test2` is standard Clojure.

```clojure
nREPL server started on port 56705 on host 127.0.0.1 - nrepl://127.0.0.1:56705
REPL-y 0.3.7, nREPL 0.2.12
Clojure 1.8.0
Java HotSpot(TM) 64-Bit Server VM 1.8.0_112-b16
    Docs: (doc function-name-here)
          (find-doc "part-of-name-here")
  Source: (source function-name-here)
 Javadoc: (javadoc java-object-or-class-here)
    Exit: Control+D or (exit) or (quit)
 Results: Stored in vars *1, *2, *3, an exception in *e

frs.core=> (use 'flatland.ordered.map)
nil
frs.core=> (def test1 (ordered-map :a (ordered-map :deck '("spark")) :b (ordered-map :deck '("flare" "bolt"))))
#'frs.core/test1
frs.core=> (def test2 {:a {:deck '("spark")} :b {:deck '("flare" "bolt")}})
#'frs.core/test2
frs.core=> (update-in test1 [:a :deck] vec)
{:a {:deck ("spark")} :b {:deck ("flare" "bolt")}}
frs.core=> (update-in test2 [:a :deck] vec)
{:a {:deck ["spark"]} :b {:deck ("flare" "bolt")}}
frs.core=> (prn (update-in test1 [:a :deck] vec))
#ordered/map ([:a #ordered/map ([:deck ("spark")])] [:b #ordered/map ([:deck ("flare" "bolt")])])
nil
frs.core=> (update test1 :a update ,,, :deck vec)
{:a {:deck ("spark")} :b {:deck ("flare" "bolt")}}
frs.core=> (update test2 :a update ,,, :deck vec)
{:a {:deck ["spark"]} :b {:deck ("flare" "bolt")}}
```

I would expect that the `update-in` or `update` of `test1` would be the same as `test2`.

Please advise?

Thanks,

Doug

# OLD MESSAGE BELOW

Hi there,

I just wanted to make you aware of a bug report I put in with specter of a problem I'm having with ordered maps and transforming them. It seems also to occur with instar. That might indicate a problem with your ordered library rather than those other two. I just don't know for sure, so I thought I'd inform you of such.

Thank you for the ordered library!

See: https://github.com/nathanmarz/specter/issues/225

Thanks,

Doug

**Comment by LispEngineer** [comment]
See comment by nathanmarz, where he suggests that your `update` methods check for equality without checking for type differences (or identity)?

nathanmarz/specter#225

**Comment by LispEngineer** [comment]
Submitted Pull Request with fix.

**Comment by LispEngineer** [comment]
Closed with 1.5.6.
